pymobiledevice3/services/dvt/instruments/process_control.py

```diff
@@ -7,6 +9,15 @@ class ProcessControl:
     def __init__(self, dvt):
         self._channel = dvt.make_channel(self.IDENTIFIER)
 
+    def send_signal(self, pid: int, sig: signal.Signals):
+        """
+        Send signal to process
+        :param pid: PID of process to send signal.
+        :param sig: SIGNAL to send
+        """
+        self._channel.sendSignal_toPid_(MessageAux().append_obj(int(sig)).append_obj(pid), expects_reply=True)
```

Comment:
* you are right.  Just wanted to support also an integer (because of pymobiledevice3 CLI) so I think the typing would better be Union[int, Signal.Signals] and then int(sig) would work on both - what do you say?
* Question - expects_reply=False would raise an exception if en errors comes (bad signal, bad pid, etc…)? If so - so I will change

Reply:
* actually i think we should use [`click.Choice`](https://click.palletsprojects.com/en/8.0.x/options/#choice-options) instead of `click.INT`. That way the user can use both `SIGKILL` and `9` as valid arguments (just like the unix `kill` command).

* yes it will. the `expects_reply` is just a protocol bit field telling the instrumentation server (`DTServiceHub`) if this message should have an ack. if it's set to true and there is no ack, then we'll have to manually call `receive_message()` after it so the message queue remains balanced.

Reply:
As for the `signal` method, in my opinion using the same argument with different types is bad. Either we receive `int` and let the caller create it however he wants, or we do the most common solution for method overloading in python - receive two optional arguments - `signal` and `signal_number` and the user can choose. I think that receiving an integer is absolutely fine (and also represents the protocol better, which is why I think that receiving the signal before the pid is better). The CLI api should be as @doronz88 said.